bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/config/RFXComRawDeviceConfiguration.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.rfxcom.internal.config;
+
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComInvalidParameterException;
+
+/**
+ * Configuration class for Raw RFXCOM device.
+ *
+ * @author James Hewitt-Thomas - Initial contribution
+ */
+public class RFXComRawDeviceConfiguration extends RFXComGenericDeviceConfiguration {
```

Comment:
Where possible we should use the null annotations, would that make sense here?

Reply:
Not sure what you mean by this one - the only method has no values that it would affect (If you mean @NonNullByDefault) and I don't think the built in Java annotations can do field validation. Are you talking about other annotations?

Reply:
Ignore me, I worked it out. Have added the annotations.

Reply:
The build is red now 🥴

Reply:
Yes, a situation I'm not used to - Eclipse says everything is fine but the CI fails to compile!

Will fix. Could be to do with the use of fields after a non null check, which fails thread safety.

Reply:
All better :+1: